src/dfx-core/src/canister/url.rs

```diff
@@ -0,0 +1,105 @@
+use url::Host::Domain;
+use url::ParseError;
+use url::Url;
+
+const MAINNET_CANDID_INTERFACE_PRINCIPAL: &str = "a4gq6-oaaaa-aaaab-qaa4q-cai";
+
+pub fn format_frontend_url(provider: &Url, canister_id: &str) -> Url {
+    let mut url = Url::clone(&provider);
+    if let Some(Domain(domain)) = url.host() {
+        if domain.ends_with("icp-api.io") || domain.ends_with("ic0.app") {
```

Comment:
Looking for substrings in domain names is always iffy, but I suspect this should be the following:
```suggestion
        if domain == "icp-api.io" || domain.ends_with(".icp-api.io") || domain == "ic0.app" || domain.ends_with(".ic0.app") {
```